### PR TITLE
If LSP server shutdown times out still restart it

### PR DESCRIPTION
**Pull request by adam-fowler**
This fixes an issue on my Windows image where the LSP server never restarts

### CODE REVIEW

File: src/sourcekit-lsp/LanguageClientManager.ts [hunk 1]
```diff
@@ -349,7 +349,10 @@ export class LanguageClientManager {
                 .then(async () => {
                     await this.setupLanguageClient(workspaceFolder?.uri);
                 })
-                .catch(reason => {
+                .catch(async reason => {
+                    if (reason.message === "Stopping the server timed out") {
+                        await this.setupLanguageClient(workspaceFolder?.uri);
+                    }
```
Comment (compnerd) [inline src/sourcekit-lsp/LanguageClientManager.ts:355]: I don't follow how this guarantees that the "old" client is terminated.  It appears to simply spawn the new instance in the case of a timeout or not.  Am I missing something?

Comment (adam-fowler) [reply src/sourcekit-lsp/LanguageClientManager.ts:355]: I've found that VSCode is not registering that the old server has been killed even though it has. It is a bug in VSCode language client really.

**Review by tristanlabelle — APPROVED** [review]
I tried this fix locally and it seems good. I also don't end up up with multiple `sourcekit-lsp.exe` processes. Thanks for fixing this!

**Comment by adam-fowler** [comment]
@swift-server-bot test this please

**Review by 0xTim — APPROVED** [review]

**Pull request merged**